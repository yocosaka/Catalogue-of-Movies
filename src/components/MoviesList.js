import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import getVisibleMovies from '../selectors/movies';
import MovieListItem from './MovieListItem';

const MoviesList = ({ movies }) => (
  <div>
    {movies.length === 0 && <p className="error-msg">Sorry, any movies have not been found. Try to search again!</p>}
    <div className="movies">
      {movies.map((movie) => {
        const {
          Title, imdbID, Year, Poster,
        } = movie;
        return (
          <MovieListItem
            Title={Title}
            imdbID={imdbID}
            Year={Year}
            Poster={Poster}
            key={imdbID}
          />
        );
      })}
    </div>
  </div>
);

const mapStateToProps = (state) => ({
  movies: getVisibleMovies(state.movies, state.filters),
});

MoviesList.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.instanceOf(Object)),
};

MoviesList.defaultProps = {
  movies: [],
};

export default connect(mapStateToProps)(MoviesList);
