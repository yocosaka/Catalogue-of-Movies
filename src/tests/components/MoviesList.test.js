import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MoviesList from '../../components/MoviesList';
import createTestStore from '../fixtures/createTestStore';

describe('MoviesList', () => {
  let store;
  beforeEach(() => {
    store = createTestStore();
  });

  test('should render the initial message before users search a movie', async () => {
    const { findByText } = render(
      <Provider store={store}>
        <MoviesList />
      </Provider>,
    );
    await findByText('Sorry, any movies have not been found. Try to search again!');
  });

  test('should match with snapshot', () => {
    const jsx = (
      <Provider store={store}>
        <BrowserRouter>
          <MoviesList />
        </BrowserRouter>
      </Provider>
    );
    const tree = (() => renderer.create(jsx).toJSON());
    expect(tree).toMatchSnapshot();
  });
});
