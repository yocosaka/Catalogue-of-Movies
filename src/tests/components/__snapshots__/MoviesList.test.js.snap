// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList should match with snapshot 1`] = `[Function]`;
